llvm/test/CodeGen/DirectX/legalize-i8.ll

```diff
@@ -110,8 +110,8 @@ define i32 @all_imm() {
 define i32 @scalar_i8_geps() {
   ; CHECK-LABEL: define i32 @scalar_i8_geps(
   ; CHECK-NEXT:    [[ALLOCA:%.*]] = alloca i32, align 4
-  ; CHECK-NEXT:    [[GEP:%.*]] = getelementptr inbounds nuw i32, ptr [[ALLOCA]], i32 0
-  ; CHECK-NEXT:    [[LOAD:%.*]] = load i32, ptr [[GEP]], align 4
+  ; CHECK-NEXT:    [[GEP:%.*]] = getelementptr inbounds nuw [1 x i32], ptr [[ALLOCA]], i32 0, i32 0
```

Comment:
I'm not sure if the validator will accept this, but I'm also unsure how to test this.

Reply:
use llc to generate DXIL https://godbolt.org/z/59KEE8rT9 feed that in to the validator

Reply:
I can't seem to get llc to produce a binary of the format that the validator is expecting.

Reply:
sorry  -filetype=obj  https://godbolt.org/z/bjEq84qME

I'm seeing the following validation failures:
```
error: Unknown part 'ISG1' found in DXIL container.
error: Unknown part 'OSG1' found in DXIL container.
error: Unknown part 'PSV0' found in DXIL container.
error: Missing part 'Runtime Data (RDAT)' required by module.
Validation failed.
```
Nothing though that would indicate an issue with GEP.